onnxruntime/test/shared_lib/test_inference.cc

```diff
@@ -2834,6 +2834,66 @@ TEST(CApiTest, ConfigureCudaArenaAndDemonstrateMemoryArenaShrinkage) {
 #ifdef USE_TENSORRT
 class CApiTensorRTTest : public testing::Test, public ::testing::WithParamInterface<std::string> {};
 
+TEST_P(CApiTensorRTTest, TestExternalCUDAStreamWithIOBinding) {
+  const auto& api = Ort::GetApi();
+  OrtTensorRTProviderOptionsV2* trt_options;
+  ASSERT_TRUE(api.CreateTensorRTProviderOptions(&trt_options) == nullptr);
+  std::unique_ptr<OrtTensorRTProviderOptionsV2, decltype(api.ReleaseTensorRTProviderOptions)> rel_trt_options(trt_options, api.ReleaseTensorRTProviderOptions);
+
+  // updating provider option with user provided compute stream
+  cudaStream_t compute_stream = nullptr;
+  void* user_compute_stream = nullptr;
+  cudaStreamCreateWithFlags(&compute_stream, cudaStreamNonBlocking);
+  ASSERT_TRUE(api.UpdateTensorRTProviderOptionsWithValue(rel_trt_options.get(), "user_compute_stream", compute_stream) == nullptr);
+  ASSERT_TRUE(api.GetTensorRTProviderOptionsByName(rel_trt_options.get(), "user_compute_stream", &user_compute_stream) == nullptr);
+  ASSERT_TRUE(user_compute_stream == (void*)compute_stream);
+
+  std::basic_string<ORTCHAR_T> model_uri = MODEL_URI;
+
+  Ort::SessionOptions session_options;
+  ASSERT_TRUE(api.SessionOptionsAppendExecutionProvider_TensorRT_V2(static_cast<OrtSessionOptions*>(session_options), rel_trt_options.get()) == nullptr);
+
+  std::unique_ptr<Ort::Session> session;
+  session.reset(new Ort::Session(*ort_env, model_uri.c_str(), session_options));
+
+  Ort::IoBinding iobindings(*session);
+
+  // input tensor on gpu
+  Ort::MemoryInfo memory_info_gpu{"Cuda", OrtDeviceAllocator, 0, OrtMemTypeDefault};
+  void* input_tensor_data = nullptr;
+  int type_size = 4;
+  std::array<float, 3 * 2> x_values = {1.0f, 2.0f, 3.0f, 4.0f, 5.0f, 6.0f};
+  std::vector<int64_t> x_shape({3, 2});
+  size_t tensor_size = x_values.size();
+  ONNXTensorElementDataType type = ONNXTensorElementDataType::ONNX_TENSOR_ELEMENT_DATA_TYPE_FLOAT;
+  assert(cudaMalloc(&input_tensor_data, tensor_size*type_size) == cudaSuccess);
+  cudaMemcpy(input_tensor_data, x_values.data(), sizeof(float) * x_values.size(), cudaMemcpyHostToDevice);
+  Ort::Value ort_input_tensor_value = Ort::Value::CreateTensor(memory_info_gpu, input_tensor_data, tensor_size * type_size,
```

Comment:
Not really point of the test, but you could also use the sessions allocator instead of raw cuda to create the `Ort::Value`. Any reason why you decided against it ? Just ease or is it not encouraged to do this ?

Reply:
yes, using ORT allocator might be better choice since it can handle releasing/freeing memory for you.
The reason i use raw cuda is that it seems some users still might use it and the test for ORT allocator is covered already.